lib/linguist/generated.rb

```diff
@@ -95,6 +96,20 @@ def minified_files?
       end
     end
 
+    # Internal: Is the blob a generated source map?
+    #
+    # Source Maps usually have .css.map or .js.map extensions. In case they
+    # are not following the name convention, detect them based on the content.
+    #
+    # Returns true or false.
+    def source_map?
+      return false unless extname.downcase == '.map'
+
+      name.downcase =~ /(\.css|\.js)\.map$/ ||         # Name convention
```

Comment:
I don't know if it matters much, but a case-insensitive regex might be better than a `.downcase`

```ruby
name =~ /(\.css|\.js)\.map$/i
```